### ISSUE TITLE
Multiple master password dialogue boxes causing problem 

### ISSUE THREAD

**Issue by dlech** [dlech/KeeAgent#227]
From official KeePass forum https://sourceforge.net/p/keepass/bugs/1730/:

>This relates to KeePass on Windows (Windows 10 for sure but I believe it is the same on other versions).
>
>Sometimes the dialogue box for entering the master password appears more than one time. The problem with this is that a second dialogue box sometimes pops up (in exactly the same place as the first) while I am typing in my master password. This causes the remainder of my password to be typed into the second dialogue box. Often I don't notice and so press Return and the password entry fails as I have only submitted the second part of my passphrase. The original dialogie box is then revealed with just the first part in (up to some unknown point).
>
>In my database I include some SSH keys and use the KeeAgent plugin to make KeePass an SSH agent. I think it is mostly SSH clients which cause this problem: sometimes they ask the agent for the private keys again and again. I cannot type my passphrase in fast enough before another request comes from the SSH client and a further dialogue box is opened. WinSCP and GitKraken for instance regularly poll the SSH agent.
>
>I hope this could be solved by KeePass ensuring that if the master password dialogue is already open when a password request (SSH agent request) comes in then a second dialogue would not be opened.
>
>Thank you.

**Comment by scp93ch** [comment]
Just confirmed it happens with the latest version of the KeeAgent plugin.

**Comment by dlech** [comment]
Latest being 0.10.1?

**Comment by scp93ch** [comment]
Yes, I did mean 0.10.1. However, the behaviour has now changed. Since the report I have upgraded KeePass to 2.39.1 (previously was 2.38).

Now it seems that if the KeePass workspace if locked and an SSH client tries to access the SSH agent then a password box is shown and the SSH client waits for the password box to close. I think that previously GitKraken (for instance) kept on creating password boxes. 

It is now much better, however if the workspace is locked, each SSH client causes a separate password box to appear (and waits separately for its password box to close). So for instance, if I do a pull in git then a password box appears. If I then try to SSH from a terminal then a second password box appears (and the first one can no longer receive focus). If I enter the password in the second box then the SSH from the terminal works but the git pull command still waits. If I then press cancel on the first password box then the git pull fails even though the KeePass workspace is now unlocked.

**Comment by scp93ch** [comment]
Actually, what I said above was slightly wrong. I just ran an ssh command in a terminal and went back to it a while later wondering why it had not logged in. The reason was that there were 30 KeePass password dialogue boxes waiting for me (none of them on the top of the stack either).
